numpy/core/src/multiarray/shape.c

```diff
@@ -683,25 +672,42 @@ PyArray_SwapAxes(PyArrayObject *ap, int a1, int a2)
                         "bad axis2 argument to swapaxes");
         return NULL;
     }
-    new_axes.ptr = dims;
-    new_axes.len = n;
 
-    for (i = 0; i < n; i++) {
-        if (i == a1) {
-            val = a2;
-        }
-        else if (i == a2) {
-            val = a1;
+    if (a1 == a2) {
+        /* Return a view of the original array */
+        PyObject *ret;
+
+        Py_INCREF(PyArray_DESCR(ap));
+        ret = PyArray_NewFromDescr(Py_TYPE(ap), PyArray_DESCR(ap), n,
```

Comment:
I used the code in `PyArray_Transpose`, which is the next function after this, as a template. We could actually eliminate this fast path altogether and let `PyArray_Transpose` handle the degenerate case also. There will be some unnecessary overhead, but perhaps simplicity is a better tradeoff than performance here.